src/components/expander.js

```diff
@@ -0,0 +1,73 @@
+var React = window.React || require('react/addons');
+var _ = require('lodash');
+var ExpanderStyles = require('../styles/expander');
+
+/*
+ * Buttons for toggling the expanded/condensed state of
+ * column items and table rows.
+ */
+
+var Expander = React.createClass({
+
+  propTypes: {
+    expanded: React.PropTypes.string,
+    onClick: React.PropTypes.func.isRequired
+  },
+
+  mixins: [ExpanderStyles],
+
+  getDefaultProps: function() {
+    return {
+      expanded: 'condensed'
+    };
+  },
+
+  getInitialState: function() {
+    // We'll use state to dump repeat clicks on the same button
+    return {
+      expandOrCondense: this.props.expanded
+    };
+  },
+
+  onExpanderClick: function(ev) {
+    ev.stopPropagation();
+    var expandOrCondense = ev.currentTarget.className;
```

Comment:
Instead of relying on looking up the className here, we could use partial application to have this method rely less on the state hidden in the (virtual) DOM. Something along the lines of:

```
onExpanderClick: function(type, ev) {
  //...
}
```

and later

```
  <button className="expanded" style={expandButton} onClick={_.partial(this.onExpanderClick, 'expanded')}>
```

That way we can keep the className as something only effecting styling rather than being indicative of UI functionality.